plugins/terraform/layer0/resource_layer0_environment.go

```diff
@@ -121,21 +143,21 @@ func resourceLayer0EnvironmentUpdate(d *schema.ResourceData, meta interface{}) e
 }
 
 func resourceLayer0EnvironmentDelete(d *schema.ResourceData, meta interface{}) error {
-	client := meta.(*Layer0Client)
+	apiClient := meta.(client.Client)
 	environmentID := d.Id()
 
-	jobID, err := client.API.DeleteEnvironment(environmentID)
+	jobID, err := apiClient.DeleteEnvironment(environmentID)
 	if err != nil {
+		return err
+	}
+
+	if _, err := client.WaitForJob(apiClient, jobID, config.DefaultTimeout); err != nil {
 		if err, ok := err.(*errors.ServerError); ok && err.Code == errors.EnvironmentDoesNotExist {
 			return nil
 		}
 
 		return err
 	}
 
-	if err := waitForJobWithContext(client, jobID); err != nil {
-		return err
-	}
-
 	return nil
 }
```

Comment:
I know you have a TODO in the ResourceProvider, but want to explicitly call out here that we do need to implement Environment Link, particularly being able to Update Environment links.